### DISCUSSION TITLE
Relation between OpenMinds, Neuroshapes, EBrain KG, HBP Altases and Nexus

### DISCUSSION THREAD

**Discussion by gavril0** [BlueBrain/nexus#2998] [Q&A]
I am teaching a course on Neuroinformatics and I would like to understand better the various HBP projects that aim at sharing data with metadata but I find the information quite confusing.

1) OpenMinds (https://github.com/HumanBrainProject/openMINDS) and Neuroshapes (https://incf.github.io/neuroshapes/) appear to be built on the top of schema.org and designed to provide meta-data. What are the differences between them?  

2) MINDS_SANDS (https://github.com/HumanBrainProject/openMINDS_SANDS) aims at providing metadata for atlases.  Is it actually used by HBP atlases (https://www.humanbrainproject.eu/en/explore-the-brain/atlases/)?  Is there a document that shows how HPB atlases metadata are specified and stored in the HBP databases? 

3) Is there publicly available datasets where or OpenMinds or Neuroshapes have been used?  Is there an example where they have been used to provide metadata at the micro-level  (metadata on single recording sessions, tasks, data processing, results, etc.?

4) EBRAINS Knowledge Graph (https://kg.ebrains.eu/) is "a metadata management system built for EBRAINS". Is there any connection to OpenMinds or Neuroshapes?   If not, how are meta-data are specified and used by EBRAIN KG ?
 

5) NEXUS seems to provide the infrastructure for EBRAIN KG (https://kg.ebrains.eu/deepdive.html) and HBP atlases. What are the other uses of NEXUS?  Does it also provide/support tools to annotate data using OpenMinds or Neuroshapes? What are these tools?

6) Neuroshapes offers a tool (https://incf.github.io/neuroshapes/docs/data-models/electrophysiology/electrophysiology.html)  to annotate neurophysiological data. Is there a publicly available data set that shows how it is used? What are the main differences with "Neurodata without border" (https://www.nwb.org/)?

7) Is there a measure of "buy-in" by the neuroscience community of the tools (OpenMinds, Neurshapes, and EBRain KG) developed by HBP ?

Thank for responding to my questions and/or providing pointers toward papers that could answer them.

Gabriel

**Comment by samuel-kerrien** [comment]
Dear Gabriel,

First of all, thanks a lot for you interest in our work and disseminating it through your Neuroinformatics course.

I will try to address as many of your questions as I can:

1. One of the key difference between OpenMINDS and Neuroshapes is that they have been built for different usecases. While Neuroshapes is primarily intended to support the Blue Brain Project's modeling and simulation of the rodent Brain, OpenMINDS is intended for sharing of Neuroscience data across the broader neuroscience community. While they are both based on some shared vocabularies (e.g. schema.org), their respective usecase dictates that they enable users to describe different aspects of Neuroscience. Some other differences are more technical, such that OpenMINDS is using JSON Schemas while Neuroshapes is using [W3C SHACL](https://www.w3.org/TR/shacl/) to constrain its metadata definition.

2. I recommend to speak with a member of the ebrains team to find out more on this topic (see below).

3. Given that we are only using Neuroshapes, I can only answer from that perspective.
At Blue Brain, we are using Blue Brain Nexus to organise the data used and produced by our scientists. Some of this data is made publicly available and I encourage you to take a look at it. You can access our deployment of Nexus and browse public data, I will now walk you through an example to illustrate with electrophysiology datasets.

   a. Open [BBP's Nexus Fusion](https://bbp.epfl.ch/nexus/web/) and browse the Studios, you do not need to login for this. Studios are a mean to present data to users and let them explore, you can find out more about Studios in the [Nexus documentation](https://bluebrainnexus.io/docs/fusion/studio.html). 
   
   b. You should now only see the list of public Studios since you are not logged in.
![all-studios](https://user-images.githubusercontent.com/3679766/146357420-7627e2eb-2f7f-4c98-bc62-24560055a86e.png)

   c. Now you can open any of these Studios, in my example I will show the `Hippocampus portal data`. As you can see on the capture below, this Studio contains two panels that expose `Neuron Morphology` and `Neuron Electrophysiology` categories. Each of these tabs presents a list of datasets. In the case of the Electrophysiology, there are 80 datasets originating from University College London (UCL). You can also see that there are 2 groups of `Neuron Morphology` originating from LNMC and UCL.
![hippocampushub-studio](https://user-images.githubusercontent.com/3679766/146357565-841ad1c8-4c40-4ea8-8568-8eb67d6daa26.png)

   d. Next, you can click on one of the table row, which represent individual datasets. In the capture below, I have selected a Neuron Electrophysiology dataset. This opens a side panel that presents more information about this dataset. Each section of this panel is called a `plugin`, it gathers information about the dataset from the Knowledge Graph and presents it in a user friendly way, I will illustrate a few of these plugins below:

   i) Electrophysiological Recording Viewer: presents the experimental result.
![plugin-ephys](https://user-images.githubusercontent.com/3679766/146358879-0eef5db3-9ce6-4056-a7c7-89c37fdc7b79.png)

   ii) Data Download: presents the data files stored in Nexus that are attached to this dataset. In this case, you can find a DAT, ABF and NWB file which you can download.
![plugin-download](https://user-images.githubusercontent.com/3679766/146358893-5fbbaac5-c467-479c-9a1a-fdcde99b7165.png)

   iii) Admin: presents a more technical view of the metadata, in this case, the JSON-LD representation of this entity in the Knowledge Graph. Please note that the entity I show to illustrate is using the Neuroshapes vocabulary to describe the dataset.
![plugin-admin](https://user-images.githubusercontent.com/3679766/146358912-b8f47875-1e02-455b-ad9a-677c86714be4.png)

   If you are interested in a comprehensive example demonstrating the use of Blue Brain Nexus to integrate data using Neuroshapes, I would recommend looking into the following MOOC on EdX: [Simulation Neuroscience](https://www.edx.org/course/simulation-neuroscience). In week 2, we details how to download data from public databases and integrate them into Blue Brain Nexus using [Nexus Forge](https://github.com/BlueBrain/nexus-forge) and viewing the data in Nexus Fusion as demonstrated above.

4. I recommend to speak with a member of the ebrains team to find out more on this topic (see below).

5. Blue Brain Nexus has been built to support Blue Brain's mission to build a biologically detailed model and simulation of the rodent brain. Now given the broad variety of neuroscience data needed to fuel such a mission, we have designed Nexus such that you can integrate any kind of data. This flexibility has also enabled several other organisations around the world to adopt and take advantage of Nexus since it is Open Source. To name a few:
   a. Human Brain Project,
   b. Krembil Center for Neuroinformatics,
   c. Research Data Connectome.

   Also please note that we have a publication describing Blue Brain Nexus in great details under review, it also describes how the organisation above are using it: 
[Blue Brain Nexus: An open, secure, scalable system for knowledge graph management and data-driven science](http://semantic-web-journal.net/system/files/swj2974.pdf)

   With respect to tools to manage data, I would recommend you to look into [Nexus Forge](https://github.com/BlueBrain/nexus-forge), its [documentation](https://nexus-forge.readthedocs.io/en/latest/) and in particular to the collection of [Notebooks](https://github.com/BlueBrain/nexus-forge/tree/master/examples/notebooks) that illustrate its usage.

6. Please take a look at the detailed answer I provided to your question 3. Now with respect to the difference between Neuroshapes and NWB, while Neuroshapes provides you a vocabulary to describe your data, NWB is a data format to store and describe your electrophysiology data. We do use them in conjunction at Blue Brain to integrate and disseminate such experimental data.

7. I recommend to speak with a member of the ebrains team to find out more on this topic (see below).

To conclude, with respect to your questions related to HBP, I am proactively contacting here the following ebrains people that can help provide further information:
   - Oliver Schmid / @olinux  
   - Andrew Davison / @apdavison 

I hope this is useful to you.
Best regards

**Reply by gavril0** [reply]
Thank you very much for the detailed and helpful answer.  I'll study the material and make contact if I need more information in January. Best regards.
